### PR TITLE
Allow toggling between full date and year only

### PR DESCRIPTION
**Pull request by randycoulman**
Add a checkbox that toggles between a date selector and a year selector.

When toggling to the year selector, default the year based on the selected date, but only if there isn’t already a year present.

The interaction design might need some tweaking.

### CODE REVIEW

File: app/presenters/generic_file_presenter.rb [hunk 1]
```diff
@@ -23,4 +24,8 @@ class GenericFilePresenter < Sufia::GenericFilePresenter
   delegate :production_names, :venue_names, :venue_full_names, :event_type_name, to: :model
 
   delegate :public?, :discoverable?, :restricted?, :private?, to: :model
+
+  def has_year_only
```
Comment (seanculver) [inline app/presenters/generic_file_presenter.rb:28]: :+1: Woof woof

File: app/views/records/edit_fields/_date_created.html.erb [hunk 2]
```diff
@@ -1,4 +1,8 @@
-<div class="form-group optional generic_file_identifier">
+<div class="form-group optional generic_file_has_year_only">
+  <%= f.input :has_year_only, as: :boolean, label: "Year Only?" %>
```
Comment (seanculver) [inline app/views/records/edit_fields/_date_created.html.erb:2]: My guess as to why you didn't add the question mark.

File: app/presenters/generic_file_presenter.rb (hunk 1 above)
Comment (randycoulman) [inline app/presenters/generic_file_presenter.rb:28]: This is a form field, so the method name has to match, right?  If I'm wrong about that, I'd be happy to make this change.

**Comment by seanculver** [comment]
:+1: Looks good, doing a good job while we are still in jQuery land.

**Pull request merged**